src/nvim/os/env.c

```diff
@@ -223,6 +223,13 @@ void init_homedir(void)
 #endif
     homedir = xstrdup(var);
   }
+  else {
+    char IObuff_signed[MAXPATHL] = "";
```

Comment:
There's no need to create a new buffer for this.  You can just use a cast in the function call -- `uv_os_homedir((char *)IObuff, ...)`.